### DISCUSSION TITLE
Can fprime-gds read json file as dictionary?

### DISCUSSION THREAD

**Discussion by kitty99099** [nasa/fprime#3058] [Q&A] [answered]
Hello,

I'm exploring [fprime hub pattern document](https://nasa.github.io/fprime/UsersGuide/best/hub-pattern.html) recently and try to replicate [fprime-hub-pattern-example](https://github.com/fprime-community/fprime-hub-pattern-example). I'm now stuck at the bottom of fprime hub pattern document as the photo.

![image](https://github.com/user-attachments/assets/bf685dcc-470f-44b6-8e18-d7946e903570)

I can successfully generate the dictionary json file

![image](https://github.com/user-attachments/assets/cf7a6017-068d-4c9a-933a-412d94b60d3e)

However, when I tried to use `fprime-gds -n --dictionary GDSDictionaryTopologyDictionary.json --ip-client --ip-address 192.168.0.XXX`, it seems like that fprime-gds can't read json file correctly. I think xml file can work correctly.

![image](https://github.com/user-attachments/assets/20c57e86-051b-42d5-82cf-968ea3346caa)

If there is any solution of this problem?

1. Any tool help to change from json file to xml file?
2. Any way that fprime-gds can read json file?
3. Any way that fpp-to-dict tool generate xml file rather than json file?

I think one of them above can solve my problem. Thank you for your time. I will appreciate your help.

**Comment by thomas-bc** [comment, marked as answer]
The JSON dictionary was added in a recent release of the GDS (`v3.5.0`). It is likely that you are using an older version. You can try to upgrade the package in your venv like such:

```
pip install -U fprime-gds==3.5.0
```

**Reply by kitty99099** [reply]
Thank you for your answer. It worked!
